File: C++/Problems/Problems/problem_2108.cpp
#include "problem_2108.h"

bool problem_2108::test() {
    std::vector<std::string> words { "abc", "car", "ada", "racecar", "cool" };

    std::string expected = "ada";

    auto result = firstPalindrome(words);

    std::cout << result << std::endl;

    return expected == result;
}

std::string problem_2108::firstPalindrome(std::vector<std::string> &words) {
    for (auto word : words)
    {
        int half = word.size() / 2;
        bool isPal = true;
        for (int i = 0; i < half; i++)
        {
            if (word[i] != word[word.size() - i - 1])
            {
                isPal = false;
                break;
            }
        }
        if (isPal)
            return word;
    }

    return "";
}

File: Python/Problems/Problems/problem2108.py
from utils import print1DArray, print2DArray
from typing import List
from collections import Counter

from .problem import Problem


class Problem2108(Problem):
    def test(self) -> bool:
        words = ["abc", "car", "ada", "racecar", "cool"]

        expected = "ada"

        result = self.firstPalindrome(words)

        print(result)

        return expected == result

    def firstPalindrome(self, words: List[str]) -> str:
        for word in words:
            half = len(word) // 2
            isPal = True
            for i in range(half):
                if word[i] != word[-(i + 1)]:
                    isPal = False
                    break
            if isPal:
                return word

        return ""